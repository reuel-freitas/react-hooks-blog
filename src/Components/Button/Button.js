export const Button = ({ onClick, description, disabled }) => (
    <button
        onClick={onClick}
        style={styles.button}
        disabled={disabled}
    >
        {description}
    </button>
)

const styles = {
    button: {
        maxWidth: 220,
        width: '100%',
        height: 30,
        padding: 3,
        borderRadius: 17
    }
}