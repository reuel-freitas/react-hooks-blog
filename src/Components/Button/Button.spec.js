import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Button } from './Button'

describe('<Button />', () => {
    it('should render the button with text', () => {
        render(<Button description="Load more posts" />)

        expect.assertions(1);

        const button = screen.getByRole('button', { name: /load more/i });
        expect(button).toBeInTheDocument();
    });

    it('should call function on button click', () => {
        const fn = jest.fn();
        render(<Button description="Load more posts" onClick={fn} />)

        const button = screen.getByRole('button', { name: /load more/i });

        userEvent.click(button);
        expect(fn).toHaveBeenCalledTimes(1);
    })

    it('should be disabled when postsLimit is reached', () => {
        render(<Button description="Load more posts" disabled={true} />)

        const button = screen.getByRole('button', { name: /load more/i });

        expect(button).toBeDisabled();
    })

    it('should be enabled when postsLimit is not reached', () => {
        render(<Button description="Load more posts" disabled={false} />)

        const button = screen.getByRole('button', { name: /load more/i });

        expect(button).toBeEnabled();
    })

    it('should match snapshot', () => {
        const {container} = render(<Button description="Load more posts" disabled={false} />)
        expect(container.firstChild).toMatchSnapshot()
    })
})