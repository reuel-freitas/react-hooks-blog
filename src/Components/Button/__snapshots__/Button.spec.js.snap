// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<button
  style="max-width: 220px; width: 100%; height: 30px; padding: 3px; border-radius: 17px;"
>
  Load more posts
</button>
`;
